### PR TITLE
[pkg/ipmi] move ioct related code to a separate file

### PR DESCRIPTION
**Pull request by pmazzini**

### CODE REVIEW

File: pkg/ipmi/ioctl.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline pkg/ipmi/ioctl.go]: can you call this ioctl_unix.go

might as well keep the naming conventions for future work.

File: pkg/ipmi/ioctl.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+// Bits and pieces from asm-generic/ioctl.h
+package ipmi
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+const (
```
Comment (rminnich) [inline pkg/ipmi/ioctl.go]: is this stuff really not defined anywhere?

File: pkg/ipmi/ioctl.go (hunk 1 above)
Comment (pmazzini) [reply pkg/ipmi/ioctl.go]: I think this is not linux specific, I think it should work on BSD as well:
https://github.com/freebsd/freebsd/blob/master/sys/dev/tdfx/tdfx_io.h#L54

I have't tried it though.

File: pkg/ipmi/ioctl.go (hunk 2 above)
Comment (pmazzini) [reply pkg/ipmi/ioctl.go]: Not that I am aware of.

File: pkg/ipmi/ioctl.go (hunk 1 above)
Comment (hugelgupf) [reply pkg/ipmi/ioctl.go]: unix suffix usually includes both linux and *bsd.

name it `ioctl_unix.go` and add a constraint like

```go
// +build aix darwin dragonfly freebsd js,wasm linux netbsd openbsd solaris
```

or whatever subset you think it'll work for.

Comment (pmazzini) [reply pkg/ipmi/ioctl.go]: I misread `_linux.go`. I'll rename it.

**Pull request merged**